doc/nrf/config_and_build/programming.rst

```diff
@@ -1,133 +1,88 @@
+For example, if you are working with an nRF9160 DK, you need to select the correct position of the :ref:`board controller <nrf9160_ug_intro>` switch **SW10** before you program the application to your development kit.
+Programming to Thingy:91 also requires a :ref:`similar step <building_pgming>`, but using a different switch (**SW2**).
 
-For more information on building and programming using the command line, see the Zephyr documentation on :ref:`zephyr:west-build-flash-debug`.
+Programming the nRF52840 Dongle
+  To program the nRF52840 Dongle instead of a development kit, follow the programming instructions in :ref:`zephyr:nrf52840dongle_nrf52840`.
 
-Optional build parameters
-=========================
+.. _programming_params:
 
-You can customize the basic ``west build`` command in a variety of ways.
-Here are some of the possible options you can use:
+Optional programming parameters
+===============================
 
-* Some applications contain configuration overlay files that enable specific features.
-  These can be added to the ``west build`` command as follows:
+You can customize the basic ``west flash`` command in a variety of ways.
 
-  .. parsed-literal::
-     :class: highlight
+.. _programming_params_no_erase:
 
-     west build -b *build_target* -- -DOVERLAY_CONFIG="overlay-feature1.conf;overlay-feature2.conf"
+Programming without ``--erase``
+  As an alternative to the recommended ``west flash --erase``, you can also clear only those flash memory pages that are to be overwritten with the new application.
+  With such approach, the old data in other areas will be retained.
 
-  See :ref:`configuration_permanent_change` and Zephyr's :ref:`zephyr:west-building-cmake-args` for more information.
-* You can include the *directory_name* parameter to build from a directory other than the application directory.
-* You can use the *build_target@board_revision* parameter to get extra devicetree overlays with new features available for a board version.
-  The *board_revision* is printed on the label of your DK, just below the PCA number.
-  For example, if you run the west build command with an additional parameter ``@1.0.0`` for nRF9160 build target, it adds the external flash on the nRF9160 DK that was available since board version 0.14.0.
-* You can :ref:`start menuconfig with the west command <configuration_temporary_change>` to configure your application.
-* You can reuse an existing build directory for building another application for another board or build target by passing ``-p=auto`` to ``west build``.
+  To erase only the areas of flash memory that are required for programming the new application, use the following command:
 
-For more information on other optional build parameters, run the ``west build -h`` help text command.
+  .. code-block:: console
 
-.. |output_files_note| replace:: For more information about files generated as output of the build process, see :ref:`app_build_output_files`.
+     west flash
```

Comment:
I also often use "west flash --recover".
Since "west flash --help" is not very useful, as it lists generic west options and not specific ones for the nRF series, I think we should list all west flash options we support, such as --erase and --recover. I do not know if there are more than that or not. Maybe this would be a good place to list them?

Reply:
Erase is documented in the lines 44-52 as the default procedure.